src/steps/AddAnother.js

```diff
@@ -105,12 +105,19 @@ class AddAnother extends Question {
     } else if (req.method === 'POST') {
       this.parse();
       this.validate();
-
       if (this.valid) {
         this.store();
-        res.redirect(this.path);
+        if (this.xhr) {
+          res.status(OK).json({ validationErrors: [] });
+        } else {
+          res.redirect(this.path);
+        }
       } else {
-        res.render(this.template, this.locals);
+        if (this.xhr) {
+          res.json(this.buildValidationErrors);
```

Comment:
I've found an issue with this. It appears that `this.buildValidationErrors` returns an array of proxies, and invoking `res.json` with it means that it will attempt to stringify it, which is currently not allowed and will throw a circular reference error even if there's no circular reference. I'm looking for a workaround.